storage/store/results_test.go

```diff
@@ -34,105 +43,119 @@ func TestResultStoreAndRetrieve(t *testing.T) {
 }
 
 func TestResultStoreTwice(t *testing.T) {
+	lockManager := storage.NewTestingLockManager()
 	dbtest.RunWithDB(t, func(t *testing.T, db storage.DB) {
 		metrics := metrics.NewNoopCollector()
 		store1 := store.NewExecutionResults(metrics, db)
 
 		result := unittest.ExecutionResultFixture()
 		blockID := unittest.IdentifierFixture()
-		err := store1.Store(result)
-		require.NoError(t, err)
 
-		err = store1.Index(blockID, result.ID())
-		require.NoError(t, err)
+		err := unittest.WithLock(t, lockManager, storage.LockIndexExecutionResult, func(lctx lockctx.Context) error {
+			return db.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
+				err := store1.BatchStore(result, rw)
+				require.NoError(t, err)
 
-		err = store1.Store(result)
+				err = store1.BatchIndex(lctx, rw, blockID, result.ID())
+				require.NoError(t, err)
+				return nil
+			})
+		})
 		require.NoError(t, err)
 
-		err = store1.Index(blockID, result.ID())
+		err = unittest.WithLock(t, lockManager, storage.LockIndexExecutionResult, func(lctx lockctx.Context) error {
+			return db.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
+				err := store1.BatchStore(result, rw)
+				require.NoError(t, err)
+
+				err = store1.BatchIndex(lctx, rw, blockID, result.ID())
+				require.NoError(t, err)
+				return nil
+			})
+		})
 		require.NoError(t, err)
 	})
 }
 
 func TestResultBatchStoreTwice(t *testing.T) {
```

Comment:
This function is algorithmically an exact duplicate of the test `TestResultStoreTwice` before. We should remove one.